core/src/poh_service.rs

```diff
@@ -85,27 +90,48 @@ impl PohService {
         }
     }
 
-    fn tick_producer(poh_recorder: Arc<Mutex<PohRecorder>>, poh_exit: &AtomicBool) {
+    fn tick_producer(
+        poh_recorder: Arc<Mutex<PohRecorder>>,
+        poh_exit: &AtomicBool,
+        target_tick_ns: u64,
+        ticks_per_slot: u64,
+    ) {
+        info!("starting with target ns: {}", target_tick_ns);
         let poh = poh_recorder.lock().unwrap().poh.clone();
         let mut now = Instant::now();
+        let mut last_metric = Instant::now();
         let mut num_ticks = 0;
         let mut num_hashes = 0;
+        let mut total_sleep_us = 0;
         loop {
             num_hashes += NUM_HASHES_PER_BATCH;
             if poh.lock().unwrap().hash(NUM_HASHES_PER_BATCH) {
                 // Lock PohRecorder only for the final hash...
                 poh_recorder.lock().unwrap().tick();
                 num_ticks += 1;
-                if num_ticks >= DEFAULT_TICKS_PER_SLOT * 2 {
```

Comment:
was just wondering why this was here 😄

Reply:
I think it was just a convenient time to report metrics. 2 slots would be roughly 1 second on the standard hardware, of course it is faster on faster hardware.